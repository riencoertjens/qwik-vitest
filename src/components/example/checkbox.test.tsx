import { createDOM } from '@builder.io/qwik/testing';
import { test, describe, expect } from 'vitest';

import { Checkbox } from './checkbox';

describe('<Checkbox />', () => {
  test(`Should render primary variant`, async () => {
    const { screen, render } = await createDOM();
    await render(<Checkbox checked={true} variant="primary" />);
    expect(screen.innerHTML).toMatchSnapshot();
  });
  
  test(`Should select / deselect (but doesn't)`, async () => {
    const { screen, render, userEvent } = await createDOM();
    await render(<Checkbox variant="action" />);

    const selector = 'input[type="checkbox"]';
    const checkbox = screen.querySelector(selector) as HTMLInputElement;

    expect(checkbox.checked).toBeFalsy();

    await userEvent(selector, 'click'); // doesn't work
    // await userEvent(checkbox, 'click'); // doesn't work
    // await userEvent('input[type="checkbox"]', 'click'); // doesn't work
    // await userEvent(screen.querySelector(selector) as HTMLInputElement, 'click'); // doesn't work

    expect(checkbox.checked).toBeTruthy();
  });
  // works!
  test(`Should select / deselect`, async () => {
    const { screen, render } = await createDOM();
    await render(<Checkbox variant="action" />);

    const selector = 'input[type="checkbox"]';
    const checkbox = screen.querySelector(selector) as HTMLInputElement;

    expect(checkbox.checked).toBeFalsy();

    checkbox.click();
    expect(checkbox.checked).toBeTruthy();

    (screen.querySelector(selector) as HTMLInputElement).click();
    expect(checkbox.checked).toBeFalsy();

    checkbox.click();
    expect(checkbox.checked).toBeTruthy();
  });
  test(`Should render indeterminate option `, async () => {
    const { screen, render } = await createDOM();
    await render(<Checkbox indeterminate />);
    expect(screen.innerHTML).toMatchSnapshot();

    const checkbox = screen.querySelector('input') as HTMLInputElement;
    expect(checkbox.indeterminate).toBeTruthy();
    
    checkbox.click();
    expect(checkbox.checked).toBeTruthy();
  });
});
