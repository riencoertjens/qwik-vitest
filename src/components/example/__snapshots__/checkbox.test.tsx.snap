// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Checkbox /> > Should render indeterminate option  1`] = `"<!--qv --><input type=\\"checkbox\\" class=\\"appearance-none border rounded-sm w-4 h-4 checked:bg-check-mark-icon-white indeterminate:bg-minus-icon-white bg-no-repeat bg-center disabled:border-bgcolor-button-disabled disabled:checked:bg-bgcolor-button-disabled disabled:indeterminate:bg-bgcolor-button-disabled checked:bg-bgcolor-button-default indeterminate:bg-bgcolor-button-default border-bordercolor-button\\"><!--/qv-->"`;

exports[`<Checkbox /> > Should render primary variant 1`] = `"<!--qv --><input type=\\"checkbox\\" class=\\"appearance-none border rounded-sm w-4 h-4 checked:bg-check-mark-icon-white indeterminate:bg-minus-icon-white bg-no-repeat bg-center disabled:border-bgcolor-button-disabled disabled:checked:bg-bgcolor-button-disabled disabled:indeterminate:bg-bgcolor-button-disabled checked:bg-bgcolor-button-default indeterminate:bg-bgcolor-button-default border-bordercolor-button\\" checked=\\"\\"><!--/qv-->"`;
